mvrx/src/main/kotlin/com/airbnb/mvrx/CoroutinesStateStore.kt

```diff
@@ -0,0 +1,112 @@
+package com.airbnb.mvrx
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.launch
+import java.util.concurrent.Executors
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+class CoroutinesStateStore<S : MvRxState>(
+    initialState: S,
+    scope: CoroutineScope
+) : MvRxStateStore<S> {
+
+    /** Channel that serves as a trigger to flush the setState and withState queues. */
+    private val flushQueuesChannel = Channel<Unit>(capacity = Channel.CONFLATED)
+    private val setStateChannel = Channel<S.() -> S>(capacity = Channel.UNLIMITED)
```

Comment:
What benefit do we get making these channels vs. queues? Seems like are justing using poll/offer

Reply:
queues aren't thread safe, while `channel.offer` is thread safe.